modules/gapi/include/opencv2/gapi/gcommon.hpp

```diff
@@ -146,14 +165,16 @@ struct GAPI_EXPORTS_W_SIMPLE GCompileArg
     GCompileArg() = default;
 
     std::string tag;
+    std::function<void(gapi::s11n::IOStream&, const util::any&)> serialize;
+    util::any arg;
 
     // FIXME: use decay in GArg/other trait-based wrapper before leg is shot!
     template<typename T, typename std::enable_if<!detail::is_compile_arg<T>::value, int>::type = 0>
     explicit GCompileArg(T &&t)
         : tag(detail::CompileArgTag<typename std::decay<T>::type>::tag())
-        , arg(t)
-    {
-    }
+        , serialize(gapi::s11n::detail::wrap_serialize<
+                        typename std::decay<T>::type, cv::GCompileArg>::serialize)
```

Comment:
not sure if decay is required at this point, I'd push it one lever lower. Also, not certain why `cv::GCompileArg` is required to be among these template parameters.

Reply:
cv::GCompileArg is removed from template arguments.
decay is put some levels lower.